cg/meta/demultiplex/demux_post_processing.py

```diff
@@ -29,7 +32,7 @@
 class DemuxPostProcessingAPI:
     """Post demultiplexing API class."""
 
-    def __init__(self, config: CGConfig):
+    def __init__(self, config: CGConfig, flow_cell_name: Optional[str] = None) -> None:
```

Comment:
I'm not entirely sure it makes sense to create the DemuxPostProcessingAPI for a specific flow cell. Maybe it does? Since we only ever create it when we are demultiplexing a single flow cell.

Reply:
future PR - but yes I agree. I think it is done for each flow cell. Mabe @henrikstranneheim can answer this ?

Reply:
I don't like this since the flow cell name is optional, I guess to be compatible with the existing code? I think we should keep it simple and just pass the name from the CLI command.

Reply:
right now the optional is indeed for compatiability. @henrikstranneheim  what do you think? Is the demuxpostprocessing flow cell centric?

Reply:
Yes it is. Always a single flow cell per demux

Reply:
What compatibility is broken if flow cell name is made mandatory?

Reply:
Not sure of that yet. The way it is now it will be backward compatible with the functions that we aim to deprecate soon